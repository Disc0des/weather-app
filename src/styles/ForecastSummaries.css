.forecast-summaries {
  display: flex;
  justify-content: space-between;

}

.forecast-summary {
  align-items: center;
  background-color: cyan;
  text-align: center;
  border-style: solid;
  border-radius: 5%;
  border-color: black;
  margin: 5px;
  padding: 20px;
}

.forecast-details_date, .forecast-summary_date, .forecast-summary_description {
  font-weight: bold;
}

.forecast-details {
  width: 20vw;
  margin: auto;
  text-align: center;
  border-style: solid;
  border-color: black;
  border-radius: 10%;
  background-color: cyan;
}