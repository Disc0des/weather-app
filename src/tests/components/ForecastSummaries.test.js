import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("ForecastSummaries", () => {
  const validProps = {
    forecasts: [
      {
        date: 1111111,
        description: "stub description 1",
        icon: 781,
        temperature: {
          min: 12,
          max: 22,
        },
      },
      {
        date: 2222222,
        description: "stub description 2",
        icon: 762,
        temperature: {
          min: 15,
          max: 25,
        },
      },
    ],
    onForecastSelect: () => {},
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummaries
        forecasts={validProps.forecasts}
        onForecastSelect={validProps.onForecastSelect}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct quantity of ForecastSummary components", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries
        forecasts={validProps.forecasts}
        onForecastSelect={validProps.onForecastSelect}
      />
    );
    expect(getAllByTestId("forecast-summary")).toHaveLength(2);
  });
});
