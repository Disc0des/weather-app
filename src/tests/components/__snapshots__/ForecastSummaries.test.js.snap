// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary_date"
      >
        Thu 1st Jan
      </div>
      <div
        class="forecast-summary_icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-tornado"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary_temperature"
      >
        22°C
      </div>
      <div
        class="forecast-summary_description"
      >
        stub description 1
      </div>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary_date"
      >
        Thu 1st Jan
      </div>
      <div
        class="forecast-summary_icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-smog"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary_temperature"
      >
        25°C
      </div>
      <div
        class="forecast-summary_description"
      >
        stub description 2
      </div>
    </div>
  </div>
</DocumentFragment>
`;
